Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -119,5 +120,70 @@ public async Task GetForecastByCityNameAsync_ReturnedForecastWeatherDTO_Success(
 
             Assert.True(new CompareLogic().Compare(expectedWeatherDto, result).AreEqual);
         }
+
+        [Fact]
+        public async Task GetWeatherByArrayCityNameAsync_ReturnedDictionaryForWeatherResponseDTO_Success()
+        {
+            var cityName2 = "Paris";
+            var cityName3 = "AАА";
+            var cityName4 = string.Empty;
+            var weatherApiDto1 = new WeatherApiDTO() { CityName = cityName, TemperatureValues = new WeatherApiTempDTO() };
+            var weatherApiDto2 = new WeatherApiDTO() { CityName = cityName2, TemperatureValues = new WeatherApiTempDTO() };
+            
+            var listCityName = new List<string>() { cityName, cityName2, cityName3, cityName4 };
+            
+            var validationFailure = new List<ValidationFailure>() { new ValidationFailure("CityName", "Test validation error")};
+            var notValidResult = new ValidationResult(validationFailure);
+            var ValidResult = new ValidationResult(new List<ValidationFailure>());
+
+            SetWeatherApiServiceSettings(_weatherApiServiceMock, weatherApiDto1);
+            SetWeatherApiServiceSettings(_weatherApiServiceMock, weatherApiDto2);
+            SetWeatherApiServiceExceptionSettings(_weatherApiServiceMock, cityName3, new Exception());
+            SetWeatherApiServiceExceptionSettings(_weatherApiServiceMock, cityName4, new  ValidationException(validationFailure));
+
+            _validator
+                 .Setup(validator => validator.ValidateAsync
+                     (It.Is<ValidationContext<ForecastWeatherRequestDTO>>(context => context.InstanceToValidate.CityName == cityName4),
+                     It.IsAny<CancellationToken>()))
+                 .ReturnsAsync(notValidResult);
+            
+            _validator
+                 .Setup(validator => validator.ValidateAsync
+                     (It.Is<ValidationContext<ForecastWeatherRequestDTO>>(
+                         context => 
+                            context.InstanceToValidate.CityName == cityName
+                            || context.InstanceToValidate.CityName == cityName2
+                            || context.InstanceToValidate.CityName == cityName3),
+                     It.IsAny<CancellationToken>()))
+                 .ReturnsAsync(ValidResult);
+
+            // Act
+            var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName);
+
+            // Assert
+            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName);
+            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName2);
+            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName3);
+            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName4);
+            Assert.True(result[true].Count() == 2 && result[false].Count() == 2);
```

Comment:
Equals

Reply:
and count check should be before contains, also you should check if result has true and false keys. It's not normal that you write result[true] contains before any checks, if the test works incorrectly it will be not failed test, but test with fallen exception out of range

Reply:
+